mimium-lang/src/compiler/parser/test.rs

```diff
@@ -240,6 +247,57 @@ fn test_fndef() {
     .into_id(0..28);
     test_string!("fn hoge(input,gue){\n input\n}", ans);
 }
+#[test]
+fn global_fnmultiple() {
+    let fnty = function!(
+        vec![Type::Unknown.into_id(), Type::Unknown.into_id()],
+        Type::Unknown.into_id()
+    );
+    let ans = letrec!(
+        "hoge",
+        Some(fnty),
+        Expr::Lambda(
+            vec![
+                TypedId {
+                    id: "input".to_symbol(),
+                    ty: Type::Unknown.into_id_with_span(8..13),
+                },
+                TypedId {
+                    id: "gue".to_symbol(),
+                    ty: Type::Unknown.into_id_with_span(14..17),
+                },
+            ],
+            None,
+            Expr::Var("input".to_symbol()).into_id(21..26),
+        )
+        .into_id(0..28),
+        Some(letrec!(
+            "hoge",
+            Some(fnty),
+            Expr::Lambda(
+                vec![
+                    TypedId {
+                        id: "input".to_symbol(),
+                        ty: Type::Unknown.into_id_with_span(8..13),
+                    },
+                    TypedId {
+                        id: "gue".to_symbol(),
+                        ty: Type::Unknown.into_id_with_span(14..17),
+                    },
+                ],
+                None,
+                Expr::Var("input".to_symbol()).into_id(21..26),
+            )
+            .into_id(0..28),
+            None
+        ))
+    );
+    test_string!(
+        "fn hoge(input,gue){\n input\n}\nfn hoge(input,gue){\n input\n}",
```

Comment:
I have not given correct span for this test result but it passes somehow. `PartialEq` for `ExprNodeId` is comparing both content and span but maybe is it not working correctly?